crates/sui-indexer-alt/src/pipeline/sequential/committer.rs

```diff
@@ -108,24 +112,56 @@ pub(super) fn committer<H: Handler + 'static>(
                         continue;
                     };
 
+                    // Determine whether we need to hold back checkpoints from being committed
+                    // because of checkpoint lag.
+                    //
+                    // TODO(amnn): Test this (depends on migrations and tempdb)
+                    let commit_hi = match (checkpoint_lag, pending.last_key_value()) {
+                        (Some(lag), None) => {
+                            debug!(pipeline = H::NAME, lag, "No pending checkpoints");
+                            if rx.is_closed() && rx.is_empty() {
+                                break;
+                            } else {
+                                continue;
+                            }
+                        }
+
+                        (Some(lag), Some((pending_hi, _))) if *pending_hi < lag => {
+                            debug!(pipeline = H::NAME, lag, pending_hi, "Priming pipeline");
+                            if rx.is_closed() && rx.is_empty() {
+                                break;
+                            } else {
+                                continue;
+                            }
+                        }
```

Comment:
iirc this tries to handle checkpoints soon after genesis, if so can we merge it to next branch and make it?
```
(Some(lag), Some((pending_hi, _))) => Some(*pending_hi.saturating_sub(lag)),
```